bindings/tbdex_wasm/src/foreign_fetch.rs

```diff
@@ -0,0 +1,167 @@
+use crate::errors::{map_err, Result};
+use http_std::{Client, FetchOptions, Method, Response};
+use std::{collections::HashMap, sync::Arc};
+use tbdex::errors::TbdexError;
+use wasm_bindgen::{prelude::wasm_bindgen, JsValue};
+
+#[wasm_bindgen]
+extern "C" {
+    #[wasm_bindgen(
+        typescript_type = "{ fetch: (url: string, options?: WasmFetchOptions) => WasmResponse }"
```

Comment:
where did we land on this? This is the way?

Reply:
This is indeed the way, but a couple items to touch up subsequently...

- There may be a slightly better way to define the `typescript_type` than inlining the attribute like this
- @diehuxx and I have talked about a slight rework with the `http-std` crate in `web5-rs` which this currently utilizes
- We need to make everything all the way down async ASAP, because right now I have a hacky workaround in typescript via worker threads